src/tests/verified_chats.rs

```diff
@@ -74,8 +74,12 @@ async fn check_verified_oneonone_chat(broken_by_classical_email: bool) {
     SystemTime::shift(std::time::Duration::from_secs(3600));
     tcm.send_recv(&bob, &alice, "Using DC again").await;
 
-    // Bob's chat is marked as verified again
-    assert_verified(&alice, &bob, ProtectionStatus::Protected).await;
+    if broken_by_classical_email {
+        // Bob's chat is marked as verified again
+        assert_verified(&alice, &bob, ProtectionStatus::Protected).await;
+    } else {
+        assert_verified(&alice, &bob, ProtectionStatus::ProtectionBroken).await;
```

Comment:
So, we have a problem here. Before the new Autocrypt key (from another Bob's device) was not saved to the db. Now it is saved and the new Peerstate overwrites the existing one with the verified key. We can't create another Peerstate because we only can have one per address. Maybe we should update the existing Peerstate so that it has the new public key, but the old verified key?

The problem is that this can be an attack on verified contacts/chats.

Reply:
Maybe the solution is to load peerstate using `Peerstate::from_addr()` if `allow_aeap` is false. Need to try

Reply:
It helps, the test is fixed.

> Before the new Autocrypt key (from another Bob's device) was not saved to the db (which is wrong)

So, i was wrong here, the new Autocrypt key was saved to the db the way i suggested above (keeping the old verified key).